File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) return false;
        
        vector<int> h(26, 0);

        for (int i = 0; i < s.length(); i ++) {
            h[(int)(s[i]) - 97] ++;
            h[(int)(t[i]) - 97] --;
        }

        for(int i = 0; i < h.size(); i ++) {
            if (h[i] != 0) return false;
        }

        return true;
    }
};

File: Data Structures & Algorithms/spiral-matrix/submission-0.py
class Solution:
    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
        res = []
        t, b = 0, len(matrix) - 1
        l, r = 0, len(matrix[0]) - 1

        while t <= b and l <= r:
            print(l, r)
            for i in range(l, r+1):
                res.append(matrix[t][i])
            t += 1

            for i in range(t, b+1):
                res.append(matrix[i][r])
            r -= 1
            
            if t <= b:
                for i in range(r, l-1, -1):
                    res.append(matrix[b][i])
                b -= 1
            
            if l <= r:
                for i in range(b, t-1, -1):
                    res.append(matrix[i][l])
                l += 1
        
        return res